### DISCUSSION TITLE
Safari doesn't handle variable color in syntax highlight correctly?

### DISCUSSION THREAD

**Discussion by birdybro** [squidfunk/mkdocs-material#4602] [Q&A] [answered]
This is a report/discussion on behalf of a user @Toryalai1 of my documentation site found here --> https://mister-devel.github.io/MkDocs_MiSTer/

I've done some customization, but not anything that I would think would make any difference between Safari and Firefox/Chrome (which I've tested on both and it works fine).

https://github.com/MiSTer-devel/MkDocs_MiSTer/issues/29 - This is the description of the issue. For some reason it's making the variable font black instead of the white I picked.

https://mister-devel.github.io/MkDocs_MiSTer/advanced/videoscaling/ - Example documentation page, it's happening in the syntax highlighted sections.

Seems like `--md-code-hl-variable-color` is just being ignored by iOS for some reason hrmmm

**Comment by squidfunk** [comment]
Thanks for reporting. If you can create a minimal reproducible example, and attach it here as a zip, I can look into it.

**Comment by birdybro** [comment]
![image](https://user-images.githubusercontent.com/16388068/200318589-d8a57259-07d3-4809-b449-518b1ebb95af.png)

![image](https://user-images.githubusercontent.com/16388068/200319092-30206efb-be98-4ac6-be22-63347e517cd7.png)

https://mister-devel.github.io/MkDocs_MiSTer/advanced/videomodes/#additional-flags

I don't know what you mean specifically. I don't have the ability to reproduce it since I don't have iOS. I talked to another person who is a web dev and he said that iOS browsers all use webkit, and this user who had the issue also saw the problem in their chrome browser, so the bug is occurring due to webkit most likely.

I asked him to look at the material for mkdocs page and the same thing was not occurring, so my only guess is my custom colors are incorrect somehow :(

https://github.com/MiSTer-devel/MkDocs_MiSTer/blob/main/docs/stylesheets/extra.css

`-md-code-hl-variable-color: var(--md-code-h1-generic-color);`

That is the color in question, which is referencing this:

`--md-code-hl-generic-color: #f8f8f28a;`

**Reply by squidfunk** [reply]
> I don't know what you mean specifically. I don't have the ability to reproduce it since I don't have iOS. I talked to another person who is a web dev and he said that iOS browsers all use webkit, and this user who had the issue also saw the problem in their chrome browser, so the bug is occurring due to webkit most likely.

I'm unable to reproduce it in Chrome.

**Reply by birdybro** [reply]
Interesting... so Chrome on iOS looked correct, but you could reproduce it in Safari on iOS?

**Comment by birdybro** [comment, marked as answer]
I get what you mean by a minimal example now, sorry, coffee just kicking in ;)

[MkDocs_MiSTer.zip](https://github.com/squidfunk/mkdocs-material/files/9951975/MkDocs_MiSTer.zip)

**Reply by squidfunk** [reply]
Your example contains too many files. Please provide a minimal example which only showcases the issue but doesn't contain other plugin and extensions. You need to remove everything that is not related.

**Reply by birdybro** [reply]
Will do! I'll try and further reduce it. Sorry!

**Reply by birdybro** [reply]
[MkDocs_MiSTer.zip](https://github.com/squidfunk/mkdocs-material/files/9972557/MkDocs_MiSTer.zip)

Is this sufficient?

**Reply by squidfunk** [reply]
Yes, thanks! Much better.

Looking at the CSS, I don't see any error on our side. In fact, you have not defined `--md-code-hl-variable-color` correctly – it points to an undefined variable called `--md-code-h1-generic-color`. I assume that you meant `--md-code-hl-generic-color` (`l` instead of `1`). Changing the `1` to `l` makes it work:

``` css
[data-md-color-scheme="MiSTer-Dark"] {
  --md-code-hl-variable-color: var(--md-code-hl-generic-color);
}
```

Thus, it's a user error. Screenshot from Safari shows that it now works as expected:

<img width="536" alt="Bildschirmfoto 2022-11-09 um 17 11 05" src="https://user-images.githubusercontent.com/932156/200881717-dfc737e4-5cd3-420e-8104-1c258ea615fc.png">

**Reply by birdybro** [reply]
Thank you for catching that! I understand it much more clearly now. Thanks for correcting this nub mistake!
